common/persistence/persistenceInterface.go

```diff
@@ -155,7 +155,9 @@ type (
 		DeleteHistoryBranch(ctx context.Context, request *InternalDeleteHistoryBranchRequest) error
 		// GetHistoryTree returns all branch information of a tree
 		GetHistoryTree(ctx context.Context, request *GetHistoryTreeRequest) (*InternalGetHistoryTreeResponse, error)
-		// GetAllHistoryTreeBranches returns all branches of all trees
+		// GetAllHistoryTreeBranches returns all branches of all trees.
+		// Note that branches may be skipped or duplicated across pages if there are branches created or deleted while
+		// paginating through results.
```

Comment:
this warning might not be needed anymore

Reply:
I think it's still possible because the tree id and branch id are randomly generated